bin/cli.js

```diff
@@ -124,30 +124,41 @@ function invoke(env) {
       `-x [${filetypes.join('|')}]`,
       'Specify the knexfile extension (default js)'
     )
-    .action(() => {
-      const type = (argv.x || 'js').toLowerCase();
-      if (filetypes.indexOf(type) === -1) {
-        exit(`Invalid filetype specified: ${type}`);
-      }
-      if (env.configuration) {
-        exit(`Error: ${env.knexfile} already exists`);
+    .action(async () => {
+      try {
+        const packageDirectory = await pkgDir(process.cwd());
+        const consumingPackageJson = require(path.resolve(
+          packageDirectory,
+          'package.json'
+        ));
+        const consumingPackageIsModule = consumingPackageJson.type === 'module';
+        const stubExtension = (argv.x || 'js').toLowerCase();
+        const stubFormatToUse =
+          stubExtension === 'js'
+            ? consumingPackageIsModule
+              ? 'mjs'
+              : 'cjs'
+            : stubExtension;
+
+        if (filetypes.indexOf(stubExtension) === -1) {
+          exit(`Invalid filetype specified: ${stubExtension}`);
+        }
+        if (env.configuration) {
+          exit(`Error: ${env.knexfile} already exists`);
+        }
+        checkLocalModule(env);
+        const stubPath = `./knexfile.${stubExtension}`;
+        const code = await fsPromised.readFile(
+          `${path.dirname(
+            env.modulePath
+          )}/lib/migrate/stub/knexfile-${stubFormatToUse}.stub`
+        );
+        await fsPromised.writeFile(stubPath, code);
+        success(color.green(`Created ${stubPath}`));
+      } catch (error) {
+        console.log(error);
+        exit();
       }
-      checkLocalModule(env);
-      const stubPath = `./knexfile.${type}`;
-      pending = fsPromised
-        .readFile(
-          path.dirname(env.modulePath) +
-            '/lib/migrate/stub/knexfile-' +
-            type +
-            '.stub'
-        )
-        .then((code) => {
-          return fsPromised.writeFile(stubPath, code);
-        })
-        .then(() => {
-          success(color.green(`Created ${stubPath}`));
-        })
-        .catch(exit);
```

Comment:
This change seems unrelated to native ESM support.  (It's just updating the code to use `async` / `await` syntax, correct?)